### PR TITLE
Migrate exit ks command

### PR DESCRIPTION
**Pull request by adamkankovsky**

### CODE REVIEW

**Comment by rvykydal** [comment]
I think we should run all kickstart tests on this one once it is ready. Maybe there are some corner cases.

**Comment by adamkankovsky** [comment]
The eDNS tests are the only tests that fail, but this is a known reported issue.

File: pyanaconda/modules/common/structures/reboot.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    @property
+    def action(self) -> Int:
+        """Requested post-installation action.
+
+        Possible values:
+            - "reboot"
+            - "poweroff"
+            - "halt"
+            - "shutdown"
```
Comment (rvykydal) [inline pyanaconda/modules/common/structures/reboot.py]: The value should be Int?

Comment (adamkankovsky) [reply pyanaconda/modules/common/structures/reboot.py]: Yes it should be Int, because usually ks constant is used which is int, but the comments are old and wrong of course

Comment (rvykydal) [reply pyanaconda/modules/common/structures/reboot.py]: Yes, I was meaning updating the comment.

File: pyanaconda/modules/runtime/runtime.py [hunk 2]
```diff
@@ -172,3 +197,79 @@ def set_eula_agreed(self, agreed):
...
+
+        :param reboot: RebootData structure.
+        """
+        self._reboot = reboot
+        self.reboot_changed.emit()
+        log.debug("Reboot configuration set to: %s", str(reboot))
+
+    def exit(self):
+        """Perform cleanup and system action based on RebootData."""
+        log.info("RuntimeService: Exit called with %s", self._reboot)
```
Comment (rvykydal) [inline pyanaconda/modules/runtime/runtime.py]: Very minor thing: maybe we don't need the "RuntimeService: " prefix given the module of origin is apparent in the log message:
`15:13:53,817 WARNING org.fedoraproject.Anaconda.Modules.Runtime:INFO:anaconda.modules.runtime.runtime:RuntimeService: Exit called with RebootData(action=2, eject=False, kexec=False)`
(to be consistent with other logging from the module)
There is one more place with the prefix in the PR. Also maybe I would unify the first letter of exit to be both lower  or uppercase.

File: pyanaconda/startup_utils.py [hunk 3]
```diff
@@ -166,7 +167,7 @@ def check_memory(anaconda, options, display_mode=None):
             print(_("Press ENTER to continue"))
             input()
 
-        ipmi_report(IPMI_ABORTED)
+        util.ipmi_abort()
```
Comment (rvykydal) [inline pyanaconda/startup_utils.py]: I faked the failing memory check to get here and hitting Enter on the screen results in Traceback:
```
Starting installer, one moment...
^[[1;1R^[[47;195Ranaconda 43.38-1.20250902102341235228.main.4.g4be339daae.fc44 for Fedora 44 (pre-release) started.
 * installation log files are stored in /tmp during the installation
 * shell is available on TTY2 and in second TMUX pane (ctrl+b, then press 2)
 * if the graphical installation interface fails to start, try again with the
   inst.text boot option to start text installation
 * when reporting a bug add logs from /tmp as separate text/plain attachments
Fedora requires 320 MB of memory to install, but you only have 2951 MB on this machine.

Press [Enter] to reboot your system.

The installation cannot continue and the system will be rebooted
Press ENTER to continue

Press ENTER to continue

Traceback (most recent call last):
  File "/usr/bin/anaconda", line 275, in <module>
    startup_utils.check_memory(anaconda, opts, display_mode=constants.DisplayModes.TUI)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.14/site-packages/pyanaconda/startup_utils.py", line 170, in check_memory
    util.ipmi_abort()
    ~~~~~~~~~~~~~~~^^
  File "/usr/lib64/python3.14/site-packages/pyanaconda/core/util.py", line 608, in ipmi_abort
    runOnErrorScripts()
    ~~~~~~~~~~~~~~~~~^^
  File "/usr/lib64/python3.14/site-packages/pyanaconda/core/util.py", line 621, in runOnErrorScripts
    onerror_task_path = scripts_proxy.RunScriptsWithTask(KS_SCRIPT_ONERROR)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.14/site-packages/dasbus/client/proxy.py", line 161, in __getattr__
    member = self._get_member(self._get_interface(name), name)
                              ~~~~~~~~~~~~~~~~~~~^^^^^^
  File "/usr/lib/python3.14/site-packages/dasbus/client/proxy.py", line 211, in _get_interface
    self._handler.specification.members
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.14/site-packages/dasbus/client/handler.py", line 212, in specification
    self._specification = self._get_specification()
                          ~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/lib/python3.14/site-packages/dasbus/client/handler.py", line 332, in _get_specification
    xml = self._call_method(
        "org.freedesktop.DBus.Introspectable",
    ...<2 lines>...
        "(s)"
    )
  File "/usr/lib/python3.14/site-packages/dasbus/client/handler.py", line 435, in _call_method
    self._message_bus.connection,
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.14/site-packages/dasbus/connection.py", line 169, in connection
    self._connection = self._get_connection()
                       ~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/lib64/python3.14/site-packages/pyanaconda/core/dbus.py", line 47, in _get_connection
    bus_address = self._find_bus_address()
  File "/usr/lib64/python3.14/site-packages/pyanaconda/core/dbus.py", line 77, in _find_bus_address
    return super()._find_bus_address()
           ~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/lib64/python3.14/site-packages/pyanaconda/core/dbus.py", line 61, in _find_bus_address
    raise ConnectionError("Can't find Anaconda bus address!")
ConnectionError: Can't find Anaconda bus address!
```
Perhaps trying to use modules too early. It could happen also in some other places.
The rebooting does not work even in current rawhide for me, but I think the traceback should be fixed. (Not sure if as a follow-up or now, can require quite some work)

Comment (rvykydal) [reply pyanaconda/startup_utils.py]: The rebooting does not work even on RHEL 10 (instead `Pane is dead` happens) so I guess we don't have to block the PR on it, but I'd create a follow-up issue to fix the traceback.

File: tests/unit_tests/pyanaconda_tests/modules/runtime/test_module_runtime.py [hunk 4]
```diff
@@ -45,7 +45,11 @@ def test_kickstart_properties(self):
                     "text",
                     "cmdline",
                     "vnc",
-                    "rdp"]
+                    "rdp",
```
Comment (KKoukiou) [inline tests/unit_tests/pyanaconda_tests/modules/runtime/test_module_runtime.py:48]: Not consistent quote usage.

**Review by KKoukiou — COMMENTED** [review]
in the commit message you have some merge conflict leftover commnet: 

Conflicts:
tests/unit_tests/pyanaconda_tests/modules/runtime/test_module_runtime.py

File: pyanaconda/startup_utils.py (hunk 3 above)
Comment (rvykydal) [reply pyanaconda/startup_utils.py]: @adamkankovsky Seems the issue is still there. I am testing with
```
diff --git a/pyanaconda/startup_utils.py b/pyanaconda/startup_utils.py
index f45c995e2a..dbf08b3daf 100644
--- a/pyanaconda/startup_utils.py
+++ b/pyanaconda/startup_utils.py
@@ -155,7 +155,7 @@ def check_memory(anaconda, options, display_mode=None):
     reason_args = {"product_name": get_product_name(),
                    "needed_ram": needed_ram,
                    "total_ram": total_ram}
-    if needed_ram > total_ram:
+    if needed_ram > total_ram or True:
         if options.liveinst:
             # pylint: disable=logging-not-lazy
             stdout_log.warning(reason % reason_args)

```
Perhaps we need to first check if modules are already available before using them in `ipmi_abort` or `runOnErrorScripts`. Or catch the exception.

Comment (rvykydal) [reply pyanaconda/startup_utils.py]: And maybe also don't try to call the scripts where we didn't do it before the change.
I think we used to call ipmi_report without running the scripts on the ^^ place?
